crates/oxc_linter/src/autofix/fixer.rs

```diff
@@ -4,23 +4,87 @@ use super::Fix;
 
 pub struct Fixer<'a> {
     source_text: &'a str,
-    fixes: Vec<Fix>,
+    fixes: Vec<Fix<'a>>,
 }
 
 impl<'a> Fixer<'a> {
-    pub fn new(source_text: &'a str, fixes: Vec<Fix>) -> Self {
+    pub fn new(source_text: &'a str, fixes: Vec<Fix<'a>>) -> Self {
         Self { source_text, fixes }
     }
 
     pub fn fix(&self) -> Cow<'a, str> {
         if self.fixes.is_empty() {
             Cow::Borrowed(self.source_text)
         } else {
-            let mut fixed_source = String::new();
-            for fix in &self.fixes {
-                fixed_source = fix.apply(self.source_text);
-            }
-            Cow::Owned(fixed_source)
+            let source_text = self.source_text;
+            let mut output = String::new();
+            // To record the position of the last fix.
+            let mut last_pos = 0;
+            self.fixes.iter().for_each(|Fix { content, span }| {
+                let start = span.start;
```

Comment:
@Boshen is there a reason `start` and `end` on `Span` are `u32`?

Reply:
https://github.com/Boshen/oxc/pull/4#discussion_r1103630760
https://github.com/Boshen/oxc/issues/9

I'll document it.